addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -1520,6 +1521,39 @@ var LinkDialog = Dialog.extend({
             class: data.classes.replace(/float-\w+/, '') + ' o_btn_preview',
         }).html((data.label && data.label.length) ? data.label : data.url);
     },
+    /**
+     * @private
+     */
+    _adaptPageAnchor: function () {
+        var self = this;
+        var urlInputValue = this.$('#o_link_dialog_url_input').val();
+        var $pageAnchor = this.$('.o_link_dialog_page_anchor');
+        var isFromWebsite = _.str.startsWith(urlInputValue, '/');
+        var isEmpty = urlInputValue.length === 0;
+        var anchorsArray = ['', '#top', '#bottom'];
+        var $selectMenu = this.$('select[name="link_anchor"]');
+
+        if (isEmpty) {
+            $selectMenu.prop('selectedIndex', 0);
+        }
+        $pageAnchor.toggleClass('d-none', !isFromWebsite);
+        if (isFromWebsite) {
+            this.$('.o_anchors_loading').removeClass('d-none');
+            $.get(window.location.origin + urlInputValue, function (response) {
+                $selectMenu.empty();
+                _.each($(response).find(':not(p).oe_structure, :not(p)[data-oe-type=html]').children(), function (el) {
+                    if (typeof el.id !== 'undefined' && el.id.length > 0) {
```

Comment:
`ed.id !== undefined`, but in fact this cannot happen, `el.id` is always a string so the condition is useless
=>
```javascript
if (el.id.length) {
```